Sure, sorry.

<details>
<summary>streamflow run  --debug repro.yml</summary>

```
(cwl-conformance-venv) (python-3.13) [curso348@login210-18 streamflow]$ streamflow run  --debug repro.yml 
2026-05-07 14:42:39.952 INFO     Processing workflow 52a89ef5-d226-43f7-9fa9-bc5dced19c14
2026-05-07 14:42:39.953 INFO     Building workflow execution plan
2026-05-07 14:42:40.056 DEBUG    Translating Workflow /
2026-05-07 14:42:40.056 DEBUG    Translating WorkflowStep /rev
2026-05-07 14:42:40.067 DEBUG    Translating CommandLineTool /rev
2026-05-07 14:42:40.067 DEBUG    Translating WorkflowStep /sorted
2026-05-07 14:42:40.078 DEBUG    Translating CommandLineTool /sorted
2026-05-07 14:42:40.171 INFO     COMPLETED building of workflow execution plan
2026-05-07 14:42:40.171 INFO     EXECUTING workflow 52a89ef5-d226-43f7-9fa9-bc5dced19c14
2026-05-07 14:42:40.223 DEBUG    Step /reverse_sort-wf-default-transformer-injector received inputs ['0']
2026-05-07 14:42:40.223 DEBUG    Step /input-injector received inputs ['0']
2026-05-07 14:42:40.225 DEBUG    Retrieving available locations for job /reverse_sort-wf-default-transformer-injector/0 on __LOCAL__.
2026-05-07 14:42:40.225 DEBUG    Available locations for job /reverse_sort-wf-default-transformer-injector/0 on __LOCAL__ are ['__LOCAL__'].
2026-05-07 14:42:40.225 DEBUG    Job /reverse_sort-wf-default-transformer-injector/0 allocated locally
2026-05-07 14:42:40.225 DEBUG    Retrieving available locations for job /input-injector/0 on __LOCAL__.
2026-05-07 14:42:40.225 DEBUG    Available locations for job /input-injector/0 on __LOCAL__ are ['__LOCAL__'].
2026-05-07 14:42:40.225 DEBUG    Job /input-injector/0 allocated locally
2026-05-07 14:42:40.226 DEBUG    COMPLETED Step __deploy__/__LOCAL__
2026-05-07 14:42:40.226 DEBUG    COMPLETED Step __deploy__/29cf8966-d3f5-485c-9aa2-bb3cb4796e3d
2026-05-07 14:42:40.226 DEBUG    COMPLETED Step __deploy__/c3ef73f4-e031-4bc7-8b02-db86529e41ba
2026-05-07 14:42:40.227 DEBUG    Job /reverse_sort-wf-default-transformer-injector/0 changed status to RUNNING
2026-05-07 14:42:40.227 DEBUG    Job /input-injector/0 changed status to RUNNING
2026-05-07 14:42:40.250 DEBUG    COMPLETED Step /reverse_sort-wf-default-transformer-injector/__schedule__
2026-05-07 14:42:40.250 DEBUG    COMPLETED Step /input-injector/__schedule__
2026-05-07 14:42:40.251 DEBUG    Job /reverse_sort-wf-default-transformer-injector/0 changed status to COMPLETED
2026-05-07 14:42:40.251 DEBUG    Step /reverse_sort-wf-default-transformer-injector received termination token with Status COMPLETED
2026-05-07 14:42:40.251 DEBUG    Job /input-injector/0 changed status to COMPLETED
2026-05-07 14:42:40.251 DEBUG    Step /reverse_sort-wf-default-transformer received inputs ['0']
2026-05-07 14:42:40.251 DEBUG    Step /input-token-transformer received inputs ['0']
2026-05-07 14:42:40.251 DEBUG    Processing /mnt/lustre/scratch/nlsas/home/ulc/cursos/curso348/streamflow/common-workflow-language-1c1f122f780075d910fdfdea7e15e46eef3c078d/v1.0/v1.0/whale.txt on location __LOCAL__/__LOCAL__.
2026-05-07 14:42:40.251 DEBUG    Step /reverse_sort-wf-default-transformer received inputs ['0']
2026-05-07 14:42:40.252 DEBUG    Step /input-injector received termination token with Status COMPLETED
2026-05-07 14:42:40.252 DEBUG    Step /reverse_sort-wf-default-transformer received inputs ['0']
2026-05-07 14:42:40.252 DEBUG    COMPLETED Step /reverse_sort-wf-default-transformer-injector
2026-05-07 14:42:40.252 DEBUG    COMPLETED Step /input-injector
2026-05-07 14:42:40.252 DEBUG    Step /rev/input-token-transformer received inputs ['0']
2026-05-07 14:42:40.252 DEBUG    Processing /mnt/lustre/scratch/nlsas/home/ulc/cursos/curso348/streamflow/common-workflow-language-1c1f122f780075d910fdfdea7e15e46eef3c078d/v1.0/v1.0/whale.txt on location __LOCAL__/__LOCAL__.
2026-05-07 14:42:40.252 DEBUG    Step /input-token-transformer received termination token with Status COMPLETED
2026-05-07 14:42:40.253 DEBUG    Step /reverse_sort-token-transformer received inputs ['0']
2026-05-07 14:42:40.253 DEBUG    Step /reverse_sort-wf-default-transformer received termination token with Status COMPLETED
2026-05-07 14:42:40.253 DEBUG    COMPLETED Step /input-token-transformer
2026-05-07 14:42:40.253 DEBUG    COMPLETED Step /reverse_sort-wf-default-transformer
2026-05-07 14:42:40.253 DEBUG    Step /rev/__transfer__/input received inputs ['0']
2026-05-07 14:42:40.253 DEBUG    Step /rev/__schedule__ received inputs ['0']
2026-05-07 14:42:40.254 DEBUG    Step /rev/input-token-transformer received termination token with Status COMPLETED
2026-05-07 14:42:40.254 DEBUG    Step /reverse_sort-token-transformer received termination token with Status COMPLETED
2026-05-07 14:42:40.254 DEBUG    Retrieving available locations for job /rev/0 on 29cf8966-d3f5-485c-9aa2-bb3cb4796e3d/debian:stretch-slim.
2026-05-07 14:42:40.254 INFO     DEPLOYING 29cf8966-d3f5-485c-9aa2-bb3cb4796e3d
2026-05-07 14:42:40.255 DEBUG    EXECUTING command singularity --version 2>&1 on __LOCAL__/__LOCAL__ 
2026-05-07 14:42:40.258 DEBUG    COMPLETED Step /rev/input-token-transformer
2026-05-07 14:42:40.258 DEBUG    COMPLETED Step /reverse_sort-token-transformer
2026-05-07 14:42:40.623 DEBUG    Using singularity-ce version 4.3.3.
2026-05-07 14:42:40.624 DEBUG    EXECUTING command singularity instance start    --bind  "/tmp/streamflow:/tmp/streamflow"     --cleanenv    --contain                       --net   --network  "none"                         docker://debian:stretch-slim a05f42ec-891b-4d76-98a7-cc28a46ed742  2>&1 on __LOCAL__/__LOCAL__ 
2026-05-07 14:42:42.096 DEBUG    EXECUTING command singularity instance list --json 2>&1 on __LOCAL__/__LOCAL__ 
2026-05-07 14:42:42.136 DEBUG    EXECUTING command nproc 2>&1 on SubprocessShell with command `singularity exec instance://a05f42ec-891b-4d76-98a7-cc28a46ed742 sh`
2026-05-07 14:42:42.193 DEBUG    EXECUTING command cat /proc/meminfo | grep MemTotal | awk '{print $2}' 2>&1 on SubprocessShell with command `singularity exec instance://a05f42ec-891b-4d76-98a7-cc28a46ed742 sh`
2026-05-07 14:42:42.197 DEBUG    Host mount points: ['/', '/proc/sys/fs/binfmt_misc', '/proc/sys/fs/binfmt_misc', '/boot', '/var/lib/nfs/rpc_pipefs', '/mnt/netapp2', '/mnt/netapp1', '/mnt/netapp1/Optcesga_FT2_RHEL7', '/mnt/lustre/hsm', '/mnt/lustre/scratch', '/misc', '/net', '/cvmfs', '/mnt/netapp2/Home_FT2', '/mnt/netapp1/Store_CESGA', '/mnt/netapp2/Store_uni', '/mnt/netapp1/Store_RES', '/mnt/netapp1/Store_CSIC', '/mnt/netapp2/COM_SAS', '/mnt/netapp2/bio_databases', '/mnt/netapp2/Store_TGU_global', '/mnt/netapp2/bio_databases_ssd', '/mnt/netapp1/COM_NLSAS', '/mnt/netapp1/Home_BD', '/mnt/netapp1/bio_databases_ssd2', '/mnt/netapp1/fs_hadoop']
2026-05-07 14:42:42.197 DEBUG    EXECUTING command singularity buildcfg | grep CONF_FILE | cut -d'=' -f2 | xargs cat | grep "^bind path" 2>&1 on 29cf8966-d3f5-485c-9aa2-bb3cb4796e3d/a05f42ec-891b-4d76-98a7-cc28a46ed742 
2026-05-07 14:42:42.234 DEBUG    EXECUTING command cat /proc/1/mountinfo 2>&1 on SubprocessShell with command `singularity exec instance://a05f42ec-891b-4d76-98a7-cc28a46ed742 sh`
2026-05-07 14:42:42.348 DEBUG    Container binds: {'/.singularity.d': '/tmp/rootfs-2579039096/root/.singularity.d', '/bin': '/tmp/rootfs-2579039096/root/bin', '/boot': '/boot/tmp/rootfs-2579039096/root/boot', '/dev': '/tmp/rootfs-2579039096/root/dev', '/etc': '/tmp/rootfs-2579039096/root/etc', '/lib': '/tmp/rootfs-2579039096/root/lib', '/lib64': '/tmp/rootfs-2579039096/root/lib64', '/media': '/tmp/rootfs-2579039096/root/media', '/mnt': '/tmp/rootfs-2579039096/root/mnt', '/opt': '/tmp/rootfs-2579039096/root/opt', '/proc': '/tmp/rootfs-2579039096/root/proc', '/root': '/tmp/rootfs-2579039096/root/root', '/run': '/tmp/rootfs-2579039096/root/run', '/sbin': '/tmp/rootfs-2579039096/root/sbin', '/srv': '/tmp/rootfs-2579039096/root/srv', '/sys': '/tmp/rootfs-2579039096/root/sys', '/usr': '/tmp/rootfs-2579039096/root/usr', '/var': '/tmp/rootfs-2579039096/root/var', '/usr/share/zoneinfo/UCT': '/usr/share/zoneinfo/Europe/Madrid', '/tmp/streamflow': '/tmp/streamflow', '/scratch': '/scratch', '/mnt/netapp2': '/mnt/netapp2', '/mnt/netapp2/Home_FT2': '/mnt/netapp2/Home_FT2', '/mnt/netapp2/Store_uni': '/mnt/netapp2/Store_uni', '/mnt/netapp2/Store_emcol': '/mnt/netapp2', '/mnt/netapp2/Store_uscfmgmm': '/mnt/netapp2', '/mnt/netapp2/Store_xumetssp': '/mnt/netapp2', '/mnt/netapp2/Store_JAL': '/mnt/netapp2', '/mnt/netapp2/Store_JAL/ngs/DRAGEN': '/mnt/netapp2', '/mnt/netapp2/Store_ulcfcefb': '/mnt/netapp2', '/mnt/netapp2/Store_uvifaaff': '/mnt/netapp2', '/mnt/netapp2/uscmg_aplic': '/mnt/netapp2', '/mnt/netapp2/Store_cseygfla': '/mnt/netapp2', '/mnt/netapp2/Store_EMM1': '/mnt/netapp2', '/mnt/netapp2/Store_cseyglmf': '/mnt/netapp2', '/mnt/netapp2/Store_jcfernandez': '/mnt/netapp2', '/mnt/netapp2/Store_pouro': '/mnt/netapp2', '/mnt/netapp2/Store_uscge': '/mnt/netapp2', '/mnt/netapp2/Store_csgpomdm': '/mnt/netapp2', '/mnt/netapp2/Store_NPQ': '/mnt/netapp2', '/mnt/netapp2/Store_uscfajcs': '/mnt/netapp2', '/mnt/netapp2/Store_FPGMX': '/mnt/netapp2', '/mnt/netapp2/COM_SAS': '/mnt/netapp2/COM_SAS', '/mnt/netapp2/Store_Tragaldabas': '/mnt/netapp2', '/mnt/netapp2/Store_IEO': '/mnt/netapp2', '/mnt/netapp2/Fibratic': '/mnt/netapp2', '/mnt/netapp2/Store_bihcxamo': '/mnt/netapp2', '/mnt/netapp2/Store_uscbopdt': '/mnt/netapp2', '/mnt/netapp2/epicovigal': '/mnt/netapp2', '/mnt/netapp2/mobilegenomes': '/mnt/netapp2', '/mnt/netapp2/Store_ulcbahnf': '/mnt/netapp2', '/mnt/netapp2/posadalab2': '/mnt/netapp2', '/mnt/netapp2/Store_uscciaep': '/mnt/netapp2', '/mnt/netapp2/Store_csbbeims': '/mnt/netapp2', '/mnt/netapp2/Store_usczanpp': '/mnt/netapp2', '/mnt/netapp2/scubacancers_wp': '/mnt/netapp2', '/mnt/netapp2/Store_uvibglta': '/mnt/netapp2', '/mnt/netapp2/Store_csiaafpr': '/mnt/netapp2', '/mnt/netapp2/Store_RADprecise': '/mnt/netapp2', '/mnt/netapp2/Store_csdvecpm': '/mnt/netapp2', '/mnt/netapp2/bio_databases': '/mnt/netapp2/bio_databases', '/mnt/netapp2/Store_csbepffa': '/mnt/netapp2', '/mnt/netapp2/Store_cscyemfs': '/mnt/netapp2', '/mnt/netapp2/Exoma_OA': '/mnt/netapp2', '/mnt/netapp2/Store_CERAINE2': '/mnt/netapp2', '/mnt/netapp2/RES_uscmg': '/mnt/netapp2', '/mnt/netapp2/Store_Componc_tmp': '/mnt/netapp2', '/mnt/netapp2/Store_Logan50': '/mnt/netapp2', '/mnt/netapp2/Store_Omica': '/mnt/netapp2', '/mnt/netapp2/Store_Omica2': '/mnt/netapp2', '/mnt/netapp2/Store_Pato': '/mnt/netapp2', '/mnt/netapp2/Store_SuPepMem': '/mnt/netapp2', '/mnt/netapp2/Store_TGU_global': '/mnt/netapp2/Store_TGU_global', '/mnt/netapp2/Store_csbbeavg': '/mnt/netapp2', '/mnt/netapp2/Store_csbepaal': '/mnt/netapp2', '/mnt/netapp2/Store_csbeppgd': '/mnt/netapp2', '/mnt/netapp2/Store_csbivarg': '/mnt/netapp2', '/mnt/netapp2/Store_csbye': '/mnt/netapp2', '/mnt/netapp2/Store_csbyegim': '/mnt/netapp2', '/mnt/netapp2/Store_csdvepfm': '/mnt/netapp2', '/mnt/netapp2/Store_csdvemgr': '/mnt/netapp2', '/mnt/netapp2/Store_csebdjgl': '/mnt/netapp2', '/mnt/netapp2/Store_csfammlc': '/mnt/netapp2', '/mnt/netapp2/Store_csgcyfde': '/mnt/netapp2', '/mnt/netapp2/Store_csgcyjgp': '/mnt/netapp2', '/mnt/netapp2/Store_csgfyaam': '/mnt/netapp2', '/mnt/netapp2/Store_csgfyirt': '/mnt/netapp2', '/mnt/netapp2/Store_mncn_peces': '/mnt/netapp2', '/mnt/netapp2/Store_strauss': '/mnt/netapp2', '/mnt/netapp2/Store_otfmxavg': '/mnt/netapp2', '/mnt/netapp2/Store_ulcctfni': '/mnt/netapp2', '/mnt/netapp2/Store_ulciiska': '/mnt/netapp2', '/mnt/netapp2/Store_ulcfcggf': '/mnt/netapp2', '/mnt/netapp2/Store_ulcqfjrg': '/mnt/netapp2', '/mnt/netapp2/Store_usccqjba': '/mnt/netapp2', '/mnt/netapp2/Store_usccqjmo': '/mnt/netapp2', '/mnt/netapp2/Store_usccqpar': '/mnt/netapp2', '/mnt/netapp2/Store_uscfafsl': '/mnt/netapp2', '/mnt/netapp2/Store_uscgepcp': '/mnt/netapp2', '/mnt/netapp2/Store_uscmgpos': '/mnt/netapp2', '/mnt/netapp2/Store_uscqojgg': '/mnt/netapp2', '/mnt/netapp2/Store_uvibemva': '/mnt/netapp2', '/mnt/netapp2/Store_uvibedfg': '/mnt/netapp2', '/mnt/netapp2/Store_uvicmepl': '/mnt/netapp2', '/mnt/netapp2/Store_uvibgavc': '/mnt/netapp2', '/mnt/netapp2/Store_uvibfnrc': '/mnt/netapp2', '/mnt/netapp2/Store_uviesmsl': '/mnt/netapp2', '/mnt/netapp2/Store_uvifibap': '/mnt/netapp2', '/mnt/netapp2/bio_databases_ssd': '/mnt/netapp2/bio_databases_ssd', '/mnt/netapp2/genomas_enfermedad': '/mnt/netapp2', '/mnt/netapp1': '/mnt/netapp1', '/mnt/netapp1/Optcesga_FT2_RHEL7': '/mnt/netapp1/Optcesga_FT2_RHEL7', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/rootfs': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/underlay/.singularity.d': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/.singularity.d', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/underlay/bin': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/bin', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/underlay/boot': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/boot', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/underlay/dev': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/dev', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/underlay/etc': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/etc', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/underlay/lib': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/lib', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/underlay/lib64': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/lib64', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/underlay/media': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/media', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/underlay/mnt': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/mnt', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/underlay/opt': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/opt', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/underlay/proc': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/proc', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/underlay/root': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/root', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/underlay/run': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/run', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/underlay/sbin': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/sbin', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/underlay/srv': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/srv', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/underlay/sys': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/sys', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/underlay/usr': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/usr', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/underlay/var': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/var', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/final/.singularity.d': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/.singularity.d', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/final/bin': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/bin', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/final/boot': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/boot', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/final/dev': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/dev', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/final/etc': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/etc', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/final/lib': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/lib', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/final/lib64': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/lib64', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/final/media': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/media', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/final/mnt': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/mnt', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/final/opt': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/opt', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/final/proc': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/proc', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/final/root': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/root', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/final/run': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/run', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/final/sbin': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/sbin', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/final/srv': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/srv', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/final/sys': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/sys', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/final/usr': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/usr', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/final/usr/share/zoneinfo/UCT': '/mnt/netapp1/Optcesga_FT2_RHEL7/usr/share/zoneinfo/Europe/Madrid', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/final/var': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/rootfs-2579039096/root/var', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/final/tmp/streamflow': '/mnt/netapp1/Optcesga_FT2_RHEL7/tmp/streamflow', '/mnt/netapp1/Optcesga_FT2_RHEL7/2025/arch/intel/icelake/easybuild/software/Core/singularity/4.3.3/var/singularity/mnt/session/final/scratch': '/mnt/netapp1/Optcesga_FT2_RHEL7/scratch', '/mnt/netapp1/Store_Puertos': '/mnt/netapp1', '/mnt/netapp1/Store_CESGA': '/mnt/netapp1/Store_CESGA', '/mnt/netapp1/Store_RES': '/mnt/netapp1/Store_RES', '/mnt/netapp1/Store_chumxrcg': '/mnt/netapp1', '/mnt/netapp1/Store_CSIC': '/mnt/netapp1/Store_CSIC', '/mnt/netapp1/Store_ulcticfl': '/mnt/netapp1', '/mnt/netapp1/Store_Seaclim': '/mnt/netapp1', '/mnt/netapp1/Store_Pendeco': '/mnt/netapp1', '/mnt/netapp1/share_bioinf': '/mnt/netapp1', '/mnt/netapp1/Store_xumetcoc': '/mnt/netapp1', '/mnt/netapp1/Store_Sirena': '/mnt/netapp1', '/mnt/netapp1/Store_olamur': '/mnt/netapp1', '/mnt/netapp1/Store_uvifimdc': '/mnt/netapp1', '/mnt/netapp1/Store_csbyejso': '/mnt/netapp1', '/mnt/netapp1/Store_ulcbagsf': '/mnt/netapp1', '/mnt/netapp1/Store_csbyestj': '/mnt/netapp1', '/mnt/netapp1/Store_Lagrima': '/mnt/netapp1', '/mnt/netapp1/Store_climate_ore_group': '/mnt/netapp1', '/mnt/netapp1/Store_ulcctaan': '/mnt/netapp1', '/mnt/netapp1/Store_csbyerfg': '/mnt/netapp1', '/mnt/netapp1/Store_EVOKE': '/mnt/netapp1', '/mnt/netapp1/Store_argos': '/mnt/netapp1', '/mnt/netapp1/Store_SpongesMicrobiome': '/mnt/netapp1', '/mnt/netapp1/Store_csbyemca': '/mnt/netapp1', '/mnt/netapp1/COM_NLSAS': '/mnt/netapp1/COM_NLSAS', '/mnt/netapp1/ECOMT': '/mnt/netapp1', '/mnt/netapp1/Home_BD': '/mnt/netapp1/Home_BD', '/mnt/netapp1/Proxecto_NOS': '/mnt/netapp1', '/mnt/netapp1/Proyectos_CESGA': '/mnt/netapp1', '/mnt/netapp1/RES_SuPepMem': '/mnt/netapp1', '/mnt/netapp1/Store_CERAINE': '/mnt/netapp1', '/mnt/netapp1/Store_ChiralPro': '/mnt/netapp1', '/mnt/netapp1/Dataspace': '/mnt/netapp1', '/mnt/netapp1/Store_Cunqa': '/mnt/netapp1', '/mnt/netapp1/Store_Grheco_xen': '/mnt/netapp1', '/mnt/netapp1/Store_IMIB': '/mnt/netapp1', '/mnt/netapp1/Store_Meteo2': '/mnt/netapp1', '/mnt/netapp1/Store_Lung_CTC': '/mnt/netapp1', '/mnt/netapp1/Store_Microbioma': '/mnt/netapp1', '/mnt/netapp1/Store_PPMCAP': '/mnt/netapp1', '/mnt/netapp1/Store_STCshared': '/mnt/netapp1', '/mnt/netapp1/Store_aphanius': '/mnt/netapp1', '/mnt/netapp1/Store_biomexp': '/mnt/netapp1', '/mnt/netapp1/Store_SnailEvoLab': '/mnt/netapp1', '/mnt/netapp1/Store_csbepgnf': '/mnt/netapp1', '/mnt/netapp1/Store_csbeeelp': '/mnt/netapp1', '/mnt/netapp1/Store_csbepjpb': '/mnt/netapp1', '/mnt/netapp1/Store_csbiybor': '/mnt/netapp1', '/mnt/netapp1/Store_csbepisa': '/mnt/netapp1', '/mnt/netapp1/Store_csbeprri': '/mnt/netapp1', '/mnt/netapp1/Store_csbyecvc': '/mnt/netapp1', '/mnt/netapp1/Store_csbyepta': '/mnt/netapp1', '/mnt/netapp1/Store_csbyemes': '/mnt/netapp1', '/mnt/netapp1/Store_cscdiica': '/mnt/netapp1', '/mnt/netapp1/Store_cscenacs': '/mnt/netapp1', '/mnt/netapp1/Store_cseeiapc': '/mnt/netapp1', '/mnt/netapp1/Store_csdvejpd': '/mnt/netapp1', '/mnt/netapp1/Store_csdblagr': '/mnt/netapp1', '/mnt/netapp1/Store_cseygsgo': '/mnt/netapp1', '/mnt/netapp1/Store_cslordgl': '/mnt/netapp1', '/mnt/netapp1/Store_cspdalcs': '/mnt/netapp1', '/mnt/netapp1/Store_cstoasgh': '/mnt/netapp1', '/mnt/netapp1/Store_csgcyjgp3': '/mnt/netapp1', '/mnt/netapp1/Store_eDNA': '/mnt/netapp1', '/mnt/netapp1/Store_dlab': '/mnt/netapp1', '/mnt/netapp1/Store_microbiota': '/mnt/netapp1', '/mnt/netapp1/Store_next_data': '/mnt/netapp1', '/mnt/netapp1/Store_iadt': '/mnt/netapp1', '/mnt/netapp1/Store_gradiant': '/mnt/netapp1', '/mnt/netapp1/Store_otomtrll': '/mnt/netapp1', '/mnt/netapp1/Store_rcc_bl': '/mnt/netapp1', '/mnt/netapp1/Store_othcxtch': '/mnt/netapp1', '/mnt/netapp1/Store_otuicacc': '/mnt/netapp1', '/mnt/netapp1/Store_ulcbmavs': '/mnt/netapp1', '/mnt/netapp1/Store_ulciaaaa': '/mnt/netapp1', '/mnt/netapp1/Store_seresco': '/mnt/netapp1', '/mnt/netapp1/Store_usccimyg': '/mnt/netapp1', '/mnt/netapp1/Store_ulcesepg': '/mnt/netapp1', '/mnt/netapp1/Store_usccisss': '/mnt/netapp1', '/mnt/netapp1/Store_usccqaan': '/mnt/netapp1', '/mnt/netapp1/Store_uscgbmrf': '/mnt/netapp1', '/mnt/netapp1/Store_uscafvpc': '/mnt/netapp1', '/mnt/netapp1/Store_uscgejfl': '/mnt/netapp1', '/mnt/netapp1/Store_uscpcsbt': '/mnt/netapp1', '/mnt/netapp1/Store_uscqorgf': '/mnt/netapp1', '/mnt/netapp1/Store_uscrdjrr': '/mnt/netapp1', '/mnt/netapp1/Store_uvibcspc': '/mnt/netapp1', '/mnt/netapp1/Store_uscmlmdl': '/mnt/netapp1', '/mnt/netapp1/Store_uvicmvzo': '/mnt/netapp1', '/mnt/netapp1/Store_uvicmlgv': '/mnt/netapp1', '/mnt/netapp1/Store_uvifijcn': '/mnt/netapp1', '/mnt/netapp1/bio_databases_ssd2': '/mnt/netapp1/bio_databases_ssd2', '/mnt/netapp1/genomes_shared': '/mnt/netapp1', '/mnt/netapp1/fs_hadoop': '/mnt/netapp1/fs_hadoop', '/mnt/netapp1/genomes_tmp': '/mnt/netapp1', '/mnt/netapp1/posadalab': '/mnt/netapp1', '/mnt/netapp1/interaction': '/mnt/netapp1', '/mnt/netapp1/scubacancers': '/mnt/netapp1', '/mnt/netapp1/quantumcesga': '/mnt/netapp1', '/mnt/netapp1/share_mugg': '/mnt/netapp1', '/mnt/netapp1/translational_oncology': '/mnt/netapp1', '/mnt/netapp_ext2/Store_Meteo': '/', '/mnt/lustre/hsm': '/mnt/lustre/hsm', '/mnt/lustre/scratch': '/mnt/lustre/scratch'}
2026-05-07 14:42:42.348 DEBUG    EXECUTING command df -aT | tail -n +2 | awk 'NF == 1 {device = $1; getline; $0 = device $0} {print $7, $2, $5}' 2>&1 on SubprocessShell with command `singularity exec instance://a05f42ec-891b-4d76-98a7-cc28a46ed742 sh`
2026-05-07 14:42:42.380 INFO     COMPLETED deployment of 29cf8966-d3f5-485c-9aa2-bb3cb4796e3d
2026-05-07 14:42:42.381 DEBUG    Available locations for job /rev/0 on 29cf8966-d3f5-485c-9aa2-bb3cb4796e3d/debian:stretch-slim are ['a05f42ec-891b-4d76-98a7-cc28a46ed742'].
2026-05-07 14:42:42.381 DEBUG    Job /rev/0 allocated on location 29cf8966-d3f5-485c-9aa2-bb3cb4796e3d/a05f42ec-891b-4d76-98a7-cc28a46ed742
2026-05-07 14:42:42.382 DEBUG    EXECUTING command test -e /tmp/streamflow/cbbe271a-877e-4d67-a13a-ae49c62e107b && readlink -f /tmp/streamflow/cbbe271a-877e-4d67-a13a-ae49c62e107b 2>&1 on SubprocessShell with command `singularity exec instance://a05f42ec-891b-4d76-98a7-cc28a46ed742 sh`
2026-05-07 14:42:42.382 DEBUG    EXECUTING command test -e /tmp/streamflow/d61383e2-5721-4efe-847f-0cdc0f2f11d4 && readlink -f /tmp/streamflow/d61383e2-5721-4efe-847f-0cdc0f2f11d4 2>&1 on SubprocessShell with command `singularity exec instance://a05f42ec-891b-4d76-98a7-cc28a46ed742 sh`
2026-05-07 14:42:42.383 DEBUG    EXECUTING command test -e /tmp/streamflow/2d80428f-20e7-405a-a6c3-1bb5a6b65f18 && readlink -f /tmp/streamflow/2d80428f-20e7-405a-a6c3-1bb5a6b65f18 2>&1 on SubprocessShell with command `singularity exec instance://a05f42ec-891b-4d76-98a7-cc28a46ed742 sh`
2026-05-07 14:42:42.383 DEBUG    EXECUTING command mkdir -m 777 -p /tmp/streamflow/cbbe271a-877e-4d67-a13a-ae49c62e107b 2>&1 on SubprocessShell with command `singularity exec instance://a05f42ec-891b-4d76-98a7-cc28a46ed742 sh`
2026-05-07 14:42:42.383 DEBUG    EXECUTING command mkdir -m 777 -p /tmp/streamflow/d61383e2-5721-4efe-847f-0cdc0f2f11d4 2>&1 on SubprocessShell with command `singularity exec instance://a05f42ec-891b-4d76-98a7-cc28a46ed742 sh`
2026-05-07 14:42:42.384 DEBUG    EXECUTING command mkdir -m 777 -p /tmp/streamflow/2d80428f-20e7-405a-a6c3-1bb5a6b65f18 2>&1 on SubprocessShell with command `singularity exec instance://a05f42ec-891b-4d76-98a7-cc28a46ed742 sh`
2026-05-07 14:42:42.385 DEBUG    EXECUTING command test -e /tmp/streamflow/cbbe271a-877e-4d67-a13a-ae49c62e107b && readlink -f /tmp/streamflow/cbbe271a-877e-4d67-a13a-ae49c62e107b 2>&1 on SubprocessShell with command `singularity exec instance://a05f42ec-891b-4d76-98a7-cc28a46ed742 sh`
2026-05-07 14:42:42.385 DEBUG    EXECUTING command test -e /tmp/streamflow/d61383e2-5721-4efe-847f-0cdc0f2f11d4 && readlink -f /tmp/streamflow/d61383e2-5721-4efe-847f-0cdc0f2f11d4 2>&1 on SubprocessShell with command `singularity exec instance://a05f42ec-891b-4d76-98a7-cc28a46ed742 sh`
2026-05-07 14:42:42.386 DEBUG    EXECUTING command test -e /tmp/streamflow/2d80428f-20e7-405a-a6c3-1bb5a6b65f18 && readlink -f /tmp/streamflow/2d80428f-20e7-405a-a6c3-1bb5a6b65f18 2>&1 on SubprocessShell with command `singularity exec instance://a05f42ec-891b-4d76-98a7-cc28a46ed742 sh`
2026-05-07 14:42:42.386 DEBUG    EXECUTING command test -e /tmp/streamflow/cbbe271a-877e-4d67-a13a-ae49c62e107b && readlink -f /tmp/streamflow/cbbe271a-877e-4d67-a13a-ae49c62e107b 2>&1 on SubprocessShell with command `singularity exec instance://a05f42ec-891b-4d76-98a7-cc28a46ed742 sh`
2026-05-07 14:42:42.387 DEBUG    EXECUTING command test -e /tmp/streamflow/d61383e2-5721-4efe-847f-0cdc0f2f11d4 && readlink -f /tmp/streamflow/d61383e2-5721-4efe-847f-0cdc0f2f11d4 2>&1 on SubprocessShell with command `singularity exec instance://a05f42ec-891b-4d76-98a7-cc28a46ed742 sh`
2026-05-07 14:42:42.388 DEBUG    EXECUTING command test -e /tmp/streamflow/2d80428f-20e7-405a-a6c3-1bb5a6b65f18 && readlink -f /tmp/streamflow/2d80428f-20e7-405a-a6c3-1bb5a6b65f18 2>&1 on SubprocessShell with command `singularity exec instance://a05f42ec-891b-4d76-98a7-cc28a46ed742 sh`
2026-05-07 14:42:42.390 DEBUG    Step /rev/__schedule__ received termination token with Status COMPLETED
2026-05-07 14:42:42.391 DEBUG    EXECUTING command test -d "/tmp/streamflow/cbbe271a-877e-4d67-a13a-ae49c62e107b/f8ca17a6-f36a-4376-8eb1-c20202d8c433/whale.txt" 2>&1 on SubprocessShell with command `singularity exec instance://a05f42ec-891b-4d76-98a7-cc28a46ed742 sh`
2026-05-07 14:42:42.391 DEBUG    COMPLETED Step /rev/__schedule__
2026-05-07 14:42:42.391 DEBUG    Extracted effective locations a05f42ec-891b-4d76-98a7-cc28a46ed742 from initial locations a05f42ec-891b-4d76-98a7-cc28a46ed742
2026-05-07 14:42:42.391 INFO     COPYING from /mnt/netapp_ext2/Store_Meteo/mnt/lustre/scratch/nlsas/home/ulc/cursos/curso348/streamflow/common-workflow-language-1c1f122f780075d910fdfdea7e15e46eef3c078d/v1.0/v1.0/whale.txt to /tmp/streamflow/cbbe271a-877e-4d67-a13a-ae49c62e107b/f8ca17a6-f36a-4376-8eb1-c20202d8c433/whale.txt on location 29cf8966-d3f5-485c-9aa2-bb3cb4796e3d/a05f42ec-891b-4d76-98a7-cc28a46ed742
2026-05-07 14:42:42.392 DEBUG    EXECUTING command ln -snf /mnt/netapp_ext2/Store_Meteo/mnt/lustre/scratch/nlsas/home/ulc/cursos/curso348/streamflow/common-workflow-language-1c1f122f780075d910fdfdea7e15e46eef3c078d/v1.0/v1.0/whale.txt /tmp/streamflow/cbbe271a-877e-4d67-a13a-ae49c62e107b/f8ca17a6-f36a-4376-8eb1-c20202d8c433/whale.txt 2>&1 on SubprocessShell with command `singularity exec instance://a05f42ec-891b-4d76-98a7-cc28a46ed742 sh`
2026-05-07 14:42:42.392 INFO     COMPLETED copy from /mnt/netapp_ext2/Store_Meteo/mnt/lustre/scratch/nlsas/home/ulc/cursos/curso348/streamflow/common-workflow-language-1c1f122f780075d910fdfdea7e15e46eef3c078d/v1.0/v1.0/whale.txt to /tmp/streamflow/cbbe271a-877e-4d67-a13a-ae49c62e107b/f8ca17a6-f36a-4376-8eb1-c20202d8c433/whale.txt on location 29cf8966-d3f5-485c-9aa2-bb3cb4796e3d/a05f42ec-891b-4d76-98a7-cc28a46ed742
2026-05-07 14:42:42.392 DEBUG    EXECUTING command test -L /tmp/streamflow/cbbe271a-877e-4d67-a13a-ae49c62e107b/f8ca17a6-f36a-4376-8eb1-c20202d8c433/whale.txt 2>&1 on SubprocessShell with command `singularity exec instance://a05f42ec-891b-4d76-98a7-cc28a46ed742 sh`
2026-05-07 14:42:42.393 DEBUG    Step /rev received inputs ['0']
2026-05-07 14:42:42.393 DEBUG    Step /rev/__transfer__/input received termination token with Status COMPLETED
2026-05-07 14:42:42.393 DEBUG    COMPLETED Step /rev/__transfer__/input
2026-05-07 14:42:42.394 DEBUG    Job /rev/0 started
2026-05-07 14:42:42.394 DEBUG    Job /rev/0 changed status to RUNNING
2026-05-07 14:42:42.394 DEBUG    Job /rev/0 inputs: {
    "input": {
        "basename": "whale.txt",
        "checksum": "sha1$327fc7aedf4f6b69a42a7c8b808dc5a7aff61376",
        "class": "File",
        "dirname": "/tmp/streamflow/cbbe271a-877e-4d67-a13a-ae49c62e107b/f8ca17a6-f36a-4376-8eb1-c20202d8c433",
        "location": "file:///tmp/streamflow/cbbe271a-877e-4d67-a13a-ae49c62e107b/f8ca17a6-f36a-4376-8eb1-c20202d8c433/whale.txt",
        "nameext": ".txt",
        "nameroot": "whale",
        "path": "/tmp/streamflow/cbbe271a-877e-4d67-a13a-ae49c62e107b/f8ca17a6-f36a-4376-8eb1-c20202d8c433/whale.txt",
        "size": 1111
    }
}
2026-05-07 14:42:42.394 INFO     EXECUTING step /rev (job /rev/0) on location 29cf8966-d3f5-485c-9aa2-bb3cb4796e3d/a05f42ec-891b-4d76-98a7-cc28a46ed742 into directory /tmp/streamflow/d61383e2-5721-4efe-847f-0cdc0f2f11d4:
rev \
	/tmp/streamflow/cbbe271a-877e-4d67-a13a-ae49c62e107b/f8ca17a6-f36a-4376-8eb1-c20202d8c433/whale.txt
2026-05-07 14:42:42.394 DEBUG    Step /rev received termination token with Status COMPLETED
2026-05-07 14:42:42.394 DEBUG    EXECUTING command cd /tmp/streamflow/d61383e2-5721-4efe-847f-0cdc0f2f11d4 && export HOME="/tmp/streamflow/d61383e2-5721-4efe-847f-0cdc0f2f11d4" && export TMPDIR="/tmp/streamflow/2d80428f-20e7-405a-a6c3-1bb5a6b65f18" && rev /tmp/streamflow/cbbe271a-877e-4d67-a13a-ae49c62e107b/f8ca17a6-f36a-4376-8eb1-c20202d8c433/whale.txt > output.txt 2>&1 on 29cf8966-d3f5-485c-9aa2-bb3cb4796e3d/a05f42ec-891b-4d76-98a7-cc28a46ed742 for job /rev/0
2026-05-07 14:42:42.394 DEBUG    EXECUTING command singularity exec --cleanenv   --contain   instance://a05f42ec-891b-4d76-98a7-cc28a46ed742 sh -c 'cd /tmp/streamflow/d61383e2-5721-4efe-847f-0cdc0f2f11d4 && export HOME="/tmp/streamflow/d61383e2-5721-4efe-847f-0cdc0f2f11d4" && export TMPDIR="/tmp/streamflow/2d80428f-20e7-405a-a6c3-1bb5a6b65f18" && rev /tmp/streamflow/cbbe271a-877e-4d67-a13a-ae49c62e107b/f8ca17a6-f36a-4376-8eb1-c20202d8c433/whale.txt > output.txt 2>&1' 2>&1 on __LOCAL__/__LOCAL__ for job /rev/0
2026-05-07 14:42:42.451 ERROR    FAILED Job /rev/0 with error:
	
Traceback (most recent call last):
  File "/mnt/lustre/scratch/nlsas/home/ulc/cursos/curso348/streamflow/streamflow/core/recovery.py", line 42, in wrapper
    await func(*args, **kwargs)
  File "/mnt/lustre/scratch/nlsas/home/ulc/cursos/curso348/streamflow/streamflow/workflow/step.py", line 743, in _execute_command
    raise WorkflowExecutionException(
        f"FAILED Job {job.name} with error:\n\t{command_output.value}"
    )
streamflow.core.exception.WorkflowExecutionException: FAILED Job /rev/0 with error:
	
2026-05-07 14:42:42.488 WARNING  Job /rev/0 failure can not be recovered. Failure manager is not enabled.
2026-05-07 14:42:42.488 ERROR    FAILED Job /rev/0 with error:
	
2026-05-07 14:42:42.488 DEBUG    Job /rev/0 changed status to FAILED
2026-05-07 14:42:42.488 DEBUG    EXECUTING command find -L "/tmp/streamflow/d61383e2-5721-4efe-847f-0cdc0f2f11d4" -type f -exec ls -ln {} \+ | awk 'BEGIN {sum=0} {sum+=$5} END {print sum}';  2>&1 on SubprocessShell with command `singularity exec instance://a05f42ec-891b-4d76-98a7-cc28a46ed742 sh`
2026-05-07 14:42:42.491 DEBUG    EXECUTING command find -L "/tmp/streamflow/2d80428f-20e7-405a-a6c3-1bb5a6b65f18" -type f -exec ls -ln {} \+ | awk 'BEGIN {sum=0} {sum+=$5} END {print sum}';  2>&1 on SubprocessShell with command `singularity exec instance://a05f42ec-891b-4d76-98a7-cc28a46ed742 sh`
2026-05-07 14:42:42.492 DEBUG    FAILED Job /rev/0 terminated
2026-05-07 14:42:42.492 DEBUG    Step /sorted/reverse-token-transformer received termination token with Status FAILED
2026-05-07 14:42:42.492 DEBUG    Step /sorted/input-token-transformer received termination token with Status FAILED
2026-05-07 14:42:42.492 DEBUG    Step /sorted/__transfer__/reverse received termination token with Status FAILED
2026-05-07 14:42:42.492 DEBUG    Step /sorted/__transfer__/input received termination token with Status FAILED
2026-05-07 14:42:42.492 DEBUG    Step /sorted/__schedule__ received termination token with Status FAILED
2026-05-07 14:42:42.492 DEBUG    Step /sorted received termination token with Status FAILED
2026-05-07 14:42:42.493 DEBUG    Step /output-collector-transformer received termination token with Status FAILED
2026-05-07 14:42:42.493 DEBUG    Step /output-collector received termination token with Status FAILED
2026-05-07 14:42:42.493 DEBUG    Step /output-collector/__schedule__ received termination token with Status FAILED
2026-05-07 14:42:42.493 INFO     FAILED Step /rev
2026-05-07 14:42:42.493 DEBUG    FAILED Step /sorted/reverse-token-transformer
2026-05-07 14:42:42.493 DEBUG    FAILED Step /sorted/input-token-transformer
2026-05-07 14:42:42.493 DEBUG    FAILED Step /sorted/__schedule__
2026-05-07 14:42:42.493 DEBUG    FAILED Step /sorted/__transfer__/reverse
2026-05-07 14:42:42.493 DEBUG    FAILED Step /sorted/__transfer__/input
2026-05-07 14:42:42.493 INFO     FAILED Step /sorted
2026-05-07 14:42:42.493 DEBUG    FAILED Step /output-collector-transformer
2026-05-07 14:42:42.494 DEBUG    FAILED Step /output-collector/__schedule__
2026-05-07 14:42:42.494 DEBUG    FAILED Step /output-collector
2026-05-07 14:42:42.494 INFO     UNDEPLOYING 29cf8966-d3f5-485c-9aa2-bb3cb4796e3d
2026-05-07 14:42:42.494 DEBUG    EXECUTING command singularity instance stop a05f42ec-891b-4d76-98a7-cc28a46ed742 2>&1 on __LOCAL__/__LOCAL__ 
2026-05-07 14:42:42.495 INFO     UNDEPLOYING c3ef73f4-e031-4bc7-8b02-db86529e41ba
2026-05-07 14:42:42.495 INFO     COMPLETED undeployment of c3ef73f4-e031-4bc7-8b02-db86529e41ba
2026-05-07 14:42:42.616 INFO     COMPLETED undeployment of 29cf8966-d3f5-485c-9aa2-bb3cb4796e3d
2026-05-07 14:42:42.630 ERROR    FAILED Workflow execution
Traceback (most recent call last):
  File "/mnt/lustre/scratch/nlsas/home/ulc/cursos/curso348/streamflow/streamflow/main.py", line 281, in main
    asyncio.run(_async_run(parsed_args))
    ~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/mnt/netapp2/Store_uni/home/ulc/cursos/curso348/micromamba/envs/python-3.13/lib/python3.13/asyncio/runners.py", line 195, in run
    return runner.run(main)
           ~~~~~~~~~~^^^^^^
  File "/mnt/netapp2/Store_uni/home/ulc/cursos/curso348/micromamba/envs/python-3.13/lib/python3.13/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^
  File "/mnt/netapp2/Store_uni/home/ulc/cursos/curso348/micromamba/envs/python-3.13/lib/python3.13/asyncio/base_events.py", line 725, in run_until_complete
    return future.result()
           ~~~~~~~~~~~~~^^
  File "/mnt/lustre/scratch/nlsas/home/ulc/cursos/curso348/streamflow/streamflow/main.py", line 187, in _async_run
    await asyncio.gather(*workflow_tasks)
  File "/mnt/lustre/scratch/nlsas/home/ulc/cursos/curso348/streamflow/streamflow/cwl/main.py", line 93, in main
    output_tokens = await executor.run()
                    ^^^^^^^^^^^^^^^^^^^^
  File "/mnt/lustre/scratch/nlsas/home/ulc/cursos/curso348/streamflow/streamflow/workflow/executor.py", line 175, in run
    raise WorkflowExecutionException("FAILED Workflow execution")
streamflow.core.exception.WorkflowExecutionException: FAILED Workflow execution
```
</details>

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -18,6 +18,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Fixed
 
+- Fix bind mount inspection in `SingularityConnector` ([#1058](https://github.com/alpha-unito/streamflow/pull/1058))
 - Fix `--wait` flag in `Helm4Connector` for Cobra parser ([#1050](https://github.com/alpha-unito/streamflow/pull/1050))
 - Fix shell reuse collision across execution locations ([#1045](https://github.com/alpha-unito/streamflow/pull/1045))
 - Fix operators in `Hardware` and `Storage` ([#1044](https://github.com/alpha-unito/streamflow/pull/1044))

--- a/streamflow/deployment/connector/container.py
+++ b/streamflow/deployment/connector/container.py
@@ -13,6 +13,7 @@ from abc import ABC, abstractmethod
 from collections.abc import MutableMapping, MutableSequence
 from contextlib import suppress
 from importlib.resources import files
+from pathlib import PurePosixPath
 from shutil import which
 from typing import Any, AsyncContextManager
 
@@ -41,6 +42,13 @@ from streamflow.deployment.connector.base import (
 )
 from streamflow.deployment.wrapper import ConnectorWrapper, get_inner_location
 from streamflow.log_handler import logger
+
+
+def _get_longest_prefix_path(path: str, paths: MutableSequence[str]) -> PurePosixPath:
+    for curr in sorted(paths, reverse=True):
+        if path.startswith(curr):
+            return PurePosixPath(curr)
+    return PurePosixPath(posixpath.sep)
 
 
 async def _get_storage_from_binds(
@@ -194,7 +202,11 @@ class ContainerConnector(ConnectorWrapper, ABC):
         source_location: ExecutionLocation | None = None,
     ) -> tuple[MutableMapping[str, Any], MutableSequence[ExecutionLocation]]:
         # Get all container mounts
-        for volume in (await self._get_instance(location.name)).volumes.values():
+        for volume in sorted(
+            (await self._get_instance(location.name)).volumes.values(),
+            reverse=True,
+            key=lambda item: item.mount_point,
+        ):
             # If volume is a bind mount
             if volume.bind is not None:
                 # If path is in a persistent volume
@@ -537,7 +549,11 @@ class ContainerConnector(ConnectorWrapper, ABC):
         return effective_locations
 
     def _get_container_path(self, instance: ContainerInstance, path: str) -> str | None:
-        for volume in instance.volumes.values():
+        for volume in sorted(
+            filter(lambda x: x.bind is not None, instance.volumes.values()),
+            reverse=True,
+            key=lambda item: item.bind,
+        ):
             if volume.bind is not None and path.startswith(volume.bind):
                 path_processor = os.path if self._wraps_local() else posixpath
                 return posixpath.join(
@@ -547,7 +563,9 @@ class ContainerConnector(ConnectorWrapper, ABC):
         return None
 
     def _get_host_path(self, instance: ContainerInstance, path: str) -> str | None:
-        for volume in instance.volumes.values():
+        for volume in sorted(
+            instance.volumes.values(), reverse=True, key=lambda item: item.mount_point
+        ):
             if volume.bind is not None and path.startswith(volume.mount_point):
                 path_processor = os.path if self._wraps_local() else posixpath
                 return path_processor.join(
@@ -1829,12 +1847,11 @@ class SingularityConnector(ContainerConnector):
             )
         # Get inner location mount points
         if self._wraps_local():
-            fs_mounts = {
-                disk.device: disk.mountpoint
+            fs_mounts = [
+                disk.mountpoint
                 for disk in psutil.disk_partitions(all=True)
                 if disk.fstype not in FS_TYPES_TO_SKIP
-                and os.access(disk.mountpoint, os.R_OK)
-            }
+            ]
         else:
             stdout, returncode = await self.connector.run(
                 location=self._get_inner_location(),
@@ -1845,19 +1862,65 @@ class SingularityConnector(ContainerConnector):
                 capture_output=True,
             )
             if returncode == 0:
-                fs_mounts = {
-                    line.split(" - ")[1].split()[1]: line.split()[4]
+                fs_mounts = [
+                    line.split()[4]
                     for line in stdout.splitlines()
                     if line.split(" - ")[1].split()[0] not in FS_TYPES_TO_SKIP
-                }
-                if logger.isEnabledFor(logging.DEBUG):
-                    logger.debug(f"Host mount points: {fs_mounts}")
+                ]
             else:
                 raise WorkflowExecutionException(
                     f"FAILED retrieving volume mounts from `/proc/1/mountinfo` "
                     f"in deployment {self.connector.deployment_name}: [{returncode}]: {stdout}"
                 )
-        # Get the list of bind mounts for the container instance
+        if logger.isEnabledFor(logging.DEBUG):
+            logger.debug(f"Host mount points: {fs_mounts}")
+
+        # Aggregate all bind mounts for the container instance
+        binds = {}
+        # 1. Extract explicit user-defined bind mounts
+        for b in self.bind or ():
+            source, dest, _ = _parse_bind(b)
+            if source != dest:
+                binds[dest] = source
+        for m in self.mount or ():
+            type_, source, dest = _parse_mount(m)
+            if type_ == "bind" and source != dest:
+                binds[dest] = source
+        # 2. Extract system-level bind mounts from the Singularity configuration
+        stdout, returncode = await self.connector.run(
+            location=location,
+            command=[
+                "singularity",
+                "buildcfg",
+                "|",
+                "grep",
+                "CONF_FILE",
+                "|",
+                "cut",
+                "-d'='",
+                "-f2",
+                "|",
+                "xargs",
+                "cat",
+                "|",
+                "grep",
+                '"^bind path"',
+            ],
+            capture_output=True,
+        )
+        if returncode == 0:
+            for line in stdout.splitlines():
+                source, dest, _ = _parse_bind(line.split("=")[1].strip())
+                # Do not override any user-defined paths
+                if source != dest and dest not in binds:
+                    binds[dest] = source
+        else:
+            logger.warning(
+                f"Impossible to retrieve singularity configuration file "
+                f"in deployment {self.connector.deployment_name}"
+            )
+        # 3. Extract dynamically resolved mount points directly from the running container.
+        # This includes the discarded cases from the previous steps when the source and destination have the same path.
         stdout, returncode = await self.run(
             location=location,
             command=[
@@ -1867,37 +1930,15 @@ class SingularityConnector(ContainerConnector):
             capture_output=True,
         )
         if returncode == 0:
-            binds = {}
-            for b in self.bind or ():
-                source, dest, _ = _parse_bind(b)
-                binds[dest] = source
-            for m in self.mount or ():
-                type_, source, dest = _parse_mount(m)
-                if type_ == "bind":
-                    binds[dest] = source
             for line in stdout.splitlines():
-                if (dst := line.split()[4]) != posixpath.sep and (
-                    fs_type := line.split(" - ")[1].split()[0]
-                ) not in FS_TYPES_TO_SKIP:
-                    mount_source = line.split(" - ")[1].split()[1]
-                    if mount_source.startswith("/dev"):
-                        host_mount = line.split()[3]
-                    elif fs_type.startswith("nfs"):
-                        host_mount = None
-                        for host_source in sorted(fs_mounts.keys(), reverse=True):
-                            if mount_source.startswith(host_source):
-                                host_mount = mount_source.split(":", 1)[1]
-                                break
-                    else:
-                        host_mount = (
-                            (os.path if self._wraps_local() else posixpath).join(
-                                fs_mounts[mount_source], line.split()[3][1:]
-                            )
-                            if mount_source in fs_mounts
-                            else None
-                        )
-                    if dst not in binds.keys() and host_mount is not None:
-                        binds[dst] = host_mount
+                if (
+                    (dst := line.split()[4]) != posixpath.sep
+                    and dst not in binds
+                    and (line.split(" - ")[1].split()[0]) not in FS_TYPES_TO_SKIP
+                ):
+                    binds[dst] = str(
+                        _get_longest_prefix_path(dst, fs_mounts) / line.split()[3][1:]
+                    )
             if logger.isEnabledFor(logging.DEBUG):
                 logger.debug(f"Container binds: {binds}")
         else:

--- a/streamflow/deployment/connector/container.py
+++ b/streamflow/deployment/connector/container.py
@@ -44,11 +44,16 @@ from streamflow.deployment.wrapper import ConnectorWrapper, get_inner_location
 from streamflow.log_handler import logger
 
 
-def _get_longest_prefix_path(path: str, paths: MutableSequence[str]) -> PurePosixPath:
-    for curr in sorted(paths, reverse=True):
-        if path.startswith(curr):
-            return PurePosixPath(curr)
-    return PurePosixPath(posixpath.sep)
+def _get_longest_prefix_path(path: str, paths: MutableSequence[str]) -> str:
+    pure_path = PurePosixPath(path)
+    return next(
+        (
+            curr
+            for curr in sorted(paths, reverse=True)
+            if pure_path.is_relative_to(curr)
+        ),
+        posixpath.sep,
+    )
 
 
 async def _get_storage_from_binds(
@@ -554,7 +559,7 @@ class ContainerConnector(ConnectorWrapper, ABC):
             reverse=True,
             key=lambda item: item.bind,
         ):
-            if volume.bind is not None and path.startswith(volume.bind):
+            if path.startswith(volume.bind):
                 path_processor = os.path if self._wraps_local() else posixpath
                 return posixpath.join(
                     volume.mount_point,
@@ -1936,8 +1941,8 @@ class SingularityConnector(ContainerConnector):
                     and dst not in binds
                     and (line.split(" - ")[1].split()[0]) not in FS_TYPES_TO_SKIP
                 ):
-                    binds[dst] = str(
-                        _get_longest_prefix_path(dst, fs_mounts) / line.split()[3][1:]
+                    binds[dst] = posixpath.join(
+                        _get_longest_prefix_path(dst, fs_mounts), line.split()[3][1:]
                     )
             if logger.isEnabledFor(logging.DEBUG):
                 logger.debug(f"Container binds: {binds}")

